src/ucs/memory/rcache.c

```diff
@@ -858,8 +855,11 @@ ucs_rcache_check_overlap(ucs_rcache_t *rcache, ucs_pgt_addr_t *start,
                 rcache, region, UCS_RCACHE_REGION_PUT_FLAG_IN_PGTABLE);
     }
 
+    /* Overlapping regions might necessitate a larger alignment, which,
+     * in turn, can result in even more overlapping regions.
+     */
     ucs_list_head_init(&region_list);
-    ucs_rcache_find_regions(rcache, *start, old_start, &region_list);
+    ucs_rcache_find_regions(rcache, *start, old_start - 1, &region_list);
```

Comment:
why changed to -1?

Reply:
ucs_rcache_find_regions() expects an inclusive end, see other uses.

Reply:
can we write the code as while loop that stops when the list of overlapping regions is empty?